hazelcast/src/hazelcast/client/serialization/json/HazelcastJsonValue.cpp

```diff
@@ -13,50 +13,37 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
-
-#include <boost/make_shared.hpp>
-
-#include "hazelcast/client/serialization/json/HazelcastJsonValue.h"
+#include "hazelcast/client/HazelcastJsonValue.h"
 
 namespace hazelcast {
     namespace client {
-        namespace serialization {
-            namespace json {
-                HazelcastJsonValue::HazelcastJsonValue() {
-                }
-
-                HazelcastJsonValue::HazelcastJsonValue(const std::string &jsonString) : jsonString(
-                        boost::make_shared<std::string>(jsonString)) {
-                }
-
-                HazelcastJsonValue::HazelcastJsonValue(std::auto_ptr<std::string> jsonString) : jsonString(jsonString) {
-                }
+        HazelcastJsonValue::HazelcastJsonValue() {
+        }
 
-                HazelcastJsonValue::~HazelcastJsonValue() {
-                }
+        HazelcastJsonValue::HazelcastJsonValue(const std::string &jsonString) : jsonString(jsonString) {
+        }
 
-                const boost::shared_ptr<std::string> HazelcastJsonValue::toJsonString() const {
-                    return jsonString;
-                }
+        HazelcastJsonValue::HazelcastJsonValue(std::auto_ptr<std::string> jsonString) : jsonString(*jsonString) {
+        }
 
-                bool HazelcastJsonValue::operator==(const HazelcastJsonValue &rhs) const {
+        HazelcastJsonValue::~HazelcastJsonValue() {
+        }
 
-                    if (jsonString == rhs.jsonString) {
-                        return true;
-                    }
+        const std::string &HazelcastJsonValue::toString() const {
+            return jsonString;
+        }
 
-                    if (!jsonString) {
-                        return false;
-                    }
+        bool HazelcastJsonValue::operator==(const HazelcastJsonValue &rhs) const {
```

Comment:
Do you plan to add additional comparison operators here also or it's based on implicit conversion to std::string only?
I mean if any friend two arg comparisons should be added here also?

Reply:
No, I don't. Can you give an example of what is in your mind? Are you asking for "<", ">" kind of operators?

Reply:
I was thinking rather about the usage of std::string or const char* as right/left operator argument, but most of the compilers should be able to deduct proper implicit conversion to HazelcastJsonValue I think. Usually the best option is to provide < (lessThen) operator only, because it works properly in every kind of the sorting for instance and other operators can be defined using this single operator only in other operator definitions. We have here direct access to std::string (json string) so user can just "expose" it in case of sorting (or similar) operations.

Reply:
yes, `const char *` is automatically converted to `const std::string` and we use it this way throughout our API. We prefer `std::string` to `const char *` in general.

Regarding the less operator, i did not put it for now. Even the equal operator is extra. I do not want to add too much. We can add later if needed.